modules/openapi-generator/src/main/resources/Java/ApiClient.mustache

```diff
@@ -626,8 +684,15 @@ public class ApiClient {
    * @return The full URL
    */
   private String buildUrl(String path, List<Pair> queryParams, List<Pair> collectionQueryParams) {
+    String baseURL;
+    if (serverIndex != null) {
+      baseURL = servers[serverIndex].URL(serverVariables);
```

Comment:
We should probably check that `serverIndex` is within the length of `servers`, as it's a user-set value and can be wrong.

Reply:
It will raise anyway.

Reply:
I agreed with @bkabrda to have a better error/exception message similar to what we've done in other generators, e.g. Python
```
        except IndexError:
            raise ValueError(
                "Invalid index {0} when selecting the host settings. "
                "Must be less than {1}".format(index, len(servers)))
```


https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/python/petstore_api/configuration.py#L316

Reply:
I have added the range check.